atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/Transaction.java

```diff
@@ -83,6 +84,17 @@
             Iterable<RangeRequest> rangeRequests);
 
     /**
+     * Same logic as {@link Transaction#getRanges(TableReference, Iterable)} but provides explicit control
+     * of parallelism. This should be favored over the former method to avoid unexpected concurrent calls
+     * hidden in the implementation.
+     */
+    @Idempotent
+    Iterable<BatchingVisitable<RowResult<byte[]>>> getRanges(
+            final TableReference tableRef,
+            Iterable<RangeRequest> rangeRequests,
+            RequestLimitedExecutorService executor);
```

Comment:
The request limiting makes sense. But should the executor be a part of the API, should we just have ``getRangesConcurrent(final TableReference tableRef, final TableReference tableRef)`` and build an executorService ourselves?

This feels like we are just shifting the concurrency control from the configuration to the API, i.e from a deployment to a product. 

I feel we have some alternatives here:

1. Should we still keep control in the config but limit it based on the requests in flight? Maybe keep a count of the getRange requests so far and have linearly/exponentially reduce the concurrency based on the current rangeRequest number?
2. Could we have a shared executor with a fixed thread pool size across all `getRangesConcurrent` requests similar to the "Atlas Cassandra KVS" pool?